qiita_db/sql_connection.py

```diff
@@ -320,158 +318,55 @@ def _sql_executor(self, sql, sql_args=None, many=False):
-                        # check if previous results needed and replace
-                        if isinstance(arg, str):
-                            result = self._regex.search(arg)
-                            if result:
-                                result_pos = int(result.group(1))
-                                sql_args[pos] = results[result_pos]
-                                clear_res = True
-                # wipe out results if needed and reset clear_res
-                if clear_res:
-                    results = []
-                    clear_res = False
-                # Fire off the SQL command
-                try:
-                    cur.execute(sql, sql_args)
-                except Exception as e:
-                    self._rollback_raise_error(queue, sql, sql_args, e)
-
-                # fetch results if available and append to results list
-                try:
-                    res = cur.fetchall()
-                except ProgrammingError as e:
-                    # ignore error if nothing to fetch
-                    pass
-                except Exception as e:
-                    self._rollback_raise_error(queue, sql, sql_args, e)
-                else:
-                    # append all results linearly
-                    results.extend(flatten(res))
-        self._connection.commit()
-        # wipe out queue since finished
-        del self.queues[queue]
-        return results
-
-    def list_queues(self):
-        """Returns list of all queue names currently in handler
-
-        Returns
-        -------
-        list
-            names of queues in handler
-        """
-        return self.queues.keys()
-
-    def create_queue(self, queue_name):
-        """Add a new queue to the connection
-
-        Parameters
-        ----------
-        queue_name : str
-            Name of the new queue
-
-        Raises
-        ------
-        KeyError
-            Queue name already exists
-        """
-        if queue_name in self.queues:
-            raise KeyError("Queue already contains %s" % queue_name)
-
-        self.queues[queue_name] = []
-
-    def add_to_queue(self, queue, sql, sql_args=None, many=False):
-        """Add an sql command to the end of a queue
+    def execute(self, sql, sql_args=None):
+        """ Executes an SQL query with no results
 
         Parameters
         ----------
-        queue : str
-            name of queue adding to
         sql : str
-            sql command to run
-        sql_args : list or tuple, optional
-            the arguments to fill sql command with
-        many : bool, optional
-            Whether or not this should be treated as an executemany command.
-            Default False
+            The SQL query
+        sql_args : tuple, list or dict, optional
+            The arguments for the SQL query
 
         Raises
         ------
-        KeyError
-            queue does not exist
+        QiitaDBExecutionError
```

Comment:
Similar to MOI, where we try and keep a logical separation even if it is being developed inside qiita.